File: WebApplication2/Application/DestinoService.cs
using WebApplication2.Domain.Entities;
using WebApplication2.Infra.Interfaces;

namespace WebApplication2.Application
{
    public class DestinoService
    {
        private readonly IDestinoRepository _repository;

        public DestinoService(IDestinoRepository repository)
        {
            _repository = repository;
        }

        public async Task<Destino> Create(Destino destino, CancellationToken token = default)
        {
            await _repository.Create(destino, token);
            await _repository.SaveChangesAsync(token);
            return destino;
        }

        public async Task<Destino?> GetById(Guid id, CancellationToken token = default)
        {
            return await _repository.GetById(id, token);
        }

        public async Task<List<Destino>> FindAll(CancellationToken token = default)
        {
            return await _repository.FindAll(token);
        }

        public async Task<Destino> Update(Guid id, Destino periferico, CancellationToken token = default)
        {
            var existente = await _repository.GetById(id, token);

            if (existente == null)
                throw new Exception("Destino não encontrado");

            existente.Nome = periferico.Nome;
            existente.ViagemId = periferico.ViagemId;

            await _repository.Update(existente, token);
            await _repository.SaveChangesAsync(token);

            return existente;
        }

        public async Task Delete(Guid id, CancellationToken token = default)
        {
            var existente = await _repository.GetById(id, token);

            if (existente == null)
                throw new Exception("Destino não encontrado");

            await _repository.Delete(existente, token);
            await _repository.SaveChangesAsync(token);
        }

    }
}

File: Front-end/frontend/src/services/destinoService.ts
import api from "../api/axios";
import type { Destino } from "../types/Destino";

export const destinoService = {
  async getAll(): Promise<Destino[]> {
    const response = await api.get("/Destino");
    return response.data;
  },

  async getById(id: string): Promise<Destino> {
    const response = await api.get(`/Destino/${id}`);
    return response.data;
  },

  async create(destino: Destino): Promise<Destino> {
    const response = await api.post("/Destino", destino);
    return response.data;
  },

  async update(id: string, destino: Destino): Promise<Destino> {
    const response = await api.put(`/Destino/${id}`, destino);
    return response.data;
  },

  async delete(id: string): Promise<void> {
    await api.delete(`/Destino/${id}`);
  },
};